ouroboros-consensus/src/Ouroboros/Consensus/Network/NodeToNode.hs

```diff
@@ -114,7 +114,6 @@ data Handlers m peer blk = Handlers {
 
     , hChainSyncServer
         :: ChainDB.Follower m blk (ChainDB.WithPoint blk (SerialisedHeader blk))
-        -> NodeToNodeVersion
```

Comment:
Marcin's PR 2090 introduced this argument to all Consensus mini protocol threads in commit 68d35f453c1b1bc1a6eb5ebc38f1bf8392f2be48.

I was looking at what the state was then, but then I realized I should just look at today's `master`. Turns out, none of our mini protocol threads (ie "handlers") use the version information.

That's based on inspection of this code and the definitions it calls to
https://github.com/input-output-hk/ouroboros-network/blob/ed24495e525931306a30de5827ea91caafbec528/ouroboros-consensus/src/Ouroboros/Consensus/Network/NodeToNode.hs#L171-L207

It looks like the NTC analogs don't even have the types for depending on the
version
https://github.com/input-output-hk/ouroboros-network/blob/ed24495e525931306a30de5827ea91caafbec528/ouroboros-consensus/src/Ouroboros/Consensus/Network/NodeToClient.hs#L88-L98

Reply:
So my thought process is: let's remove c6764f7b6c11d020c7bc160ecf4bb31ac6dec7e4 from this PR, open a separate Issue, and ping Marcin on it. Without hearing his thoughts yet, my intuition is [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) says we should remove the version from all the NTN handlers.

Reply:
I have similar intuition. But if we remove `c6764f7b6c11d020c7bc160ecf4bb31ac6dec7e4` we will also have to remove `9388a549cb0ac01caf4fe643fd022e949c546556` - if we keep the version in `chainSyncHeadersServer` then it simply makes no sense of removing it in the next PR, because `chainSyncHeadersServer` gives nice facade for its usage in `Handlers`.

Maybe it's worth having the discussion about the version numbers being removed from `Handelres` (where they are not needed) now - in this PR?

Reply:
follow up after meeting with @nfrisby - we will keep this commit but we will not remove version argument from `Handlers`